### PR TITLE
[ARCTIC-1401][Improvement]: Arctic Flink needs to organize the community-related code into a separate module

### LINKED ISSUE #1401: [Improvement]: Arctic Flink needs to organize the community-related code into a separate module
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

To make the code structure of the flink module clearer, code from dependent external projects needs to be organized into a separate module.

**Original module structure**
```
|-- flink v1.12
        |-- flink
                |-- com.netease.arctic.flink
                |-- org.apache
                           |-- flink.connector.pulsar
                           |-- iceberg.flink
        |-- flink-format
        |-- flink-runtime
```

**Optimised module structure**
```
|-- flink v1.12
        |-- flink
                |-- com.netease.arctic.flink
        |-- flink-format
        |-- flink-runtime
        |-- arctic-flink-1.12-pulsar-bridge
                |-- org.apache.flink.connector.pulsar
        |-- arctic-flink-1.12-iceberg-bridge
                |-- org.apache.iceberg.flink
```

### How should we improve?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by lklhdu**
## Why are the changes needed?

fix #1401

## Brief change log

  - *move pulsar and iceberg related code to a separate module*
  - *standardize the name format of Flink module unit test to the Test prefix*
  - *The flink 1.12, 1.14, and 1.15 versions are affected.*

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink/v1.12/iceberg/pom.xml [hunk 1]
```diff
@@ -0,0 +1,561 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>arctic-parent</artifactId>
+        <groupId>com.netease.arctic</groupId>
+        <version>0.5.0-SNAPSHOT</version>
+        <relativePath>../../../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>iceberg</artifactId>
```
Comment (zhoujinsong) [inline flink/v1.12/iceberg/pom.xml:13]: The artifact id should be more precise, like `arctic-flink-1.12-iceberg-bridge`.

File: flink/v1.12/pulsar/pom.xml [hunk 2]
```diff
@@ -0,0 +1,561 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>arctic-parent</artifactId>
+        <groupId>com.netease.arctic</groupId>
+        <version>0.5.0-SNAPSHOT</version>
+        <relativePath>../../../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>pulsar</artifactId>
```
Comment (zhoujinsong) [inline flink/v1.12/pulsar/pom.xml:13]: The artifact id should be more precise, like `arctic-flink-1.12-pulsar-bridge`.

File: flink/v1.12/arctic-flink-iceberg-bridge/pom.xml [hunk 3]
```diff
@@ -0,0 +1,561 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>arctic-parent</artifactId>
+        <groupId>com.netease.arctic</groupId>
+        <version>0.5.0-SNAPSHOT</version>
+        <relativePath>../../../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>arctic-flink-iceberg-bridge</artifactId>
```
Comment (zhoujinsong) [inline flink/v1.12/arctic-flink-iceberg-bridge/pom.xml:13]: Add Flink version into artifact id to help identify the difference flink version.
`arctic-flink-1.12-iceberg-bridge`

File: flink/v1.12/arctic-flink-iceberg-bridge/pom.xml [hunk 4]
```diff
@@ -0,0 +1,561 @@
...
+    <name>Arctic Project Flink 1.12 opensource iceberg</name>
```
Comment (zhoujinsong) [inline flink/v1.12/arctic-flink-iceberg-bridge/pom.xml:14]: Capitalization is required for module name like `Arctic Project Flink 1.12 Iceberg Bridge`

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Review by YesOrNo828 — APPROVED** [review]
LGTM
